variables.tf

```diff
@@ -129,6 +198,20 @@ variable "public_network_access_enabled" {
   description = "Whether public network access is allowed for the Cognitive Account. Defaults to `false`."
 }
 
+variable "storage" {
+  type = object({
+    storage_account_id = string
+    identity_client_id = optional(string)
+  })
+  default     = null
```

Comment:
As `storage` might be used with `for_each` I would recommend `default = []`, and `nullable = false` here.